optuna/visualization/matplotlib/_slice.py

```diff
@@ -129,35 +130,61 @@ def _generate_slice_subplot(
 ) -> "PathCollection":
     ax.set(xlabel=subplot_info.param_name, ylabel=target_name)
     scale = None
+
+    feasible = _PlotValues([], [], [])
+    infeasible = _PlotValues([], [], [])
+    for x, y, num, c in zip(
+        subplot_info.x, subplot_info.y, subplot_info.trial_numbers, subplot_info.constraints
+    ):
+        if x is not None or x != "None" or y is not None or y != "None":
+            if c:
+                feasible.x.append(x)
+                feasible.y.append(y)
+                feasible.trial_numbers.append(num)
+            else:
+                infeasible.x.append(x)
+                infeasible.y.append(y)
+                infeasible.trial_numbers.append(num)
     if subplot_info.is_log:
         ax.set_xscale("log")
         scale = "log"
     if subplot_info.is_numerical:
-        x_values = subplot_info.x
-        y_values = subplot_info.y
-        c_values = subplot_info.trial_numbers
+        feasible_x = feasible.x
+        feasible_y = feasible.y
+        feasible_c = feasible.trial_numbers
+        infeasible_x = infeasible.x
+        infeasible_y = infeasible.y
     else:
-        x_values = []
-        y_values = []
-        c_values = []
-        assert subplot_info.x_labels is not None
-        points_dict = defaultdict(list)
-        for x, y, number in zip(subplot_info.x, subplot_info.y, subplot_info.trial_numbers):
-            points_dict[x].append((y, number))
-        for x_label in subplot_info.x_labels:
-            for y, number in points_dict[x_label]:
-                x_values.append(str(x_label))
-                y_values.append(y)
-                c_values.append(number)
+        feasible_x, feasible_y, feasible_c = _get_categorical_plot_values(subplot_info, feasible)
+        infeasible_x, infeasible_y, _ = _get_categorical_plot_values(subplot_info, infeasible)
         scale = "categorical"
-    xlim = _calc_lim_with_padding(x_values, padding_ratio, scale)
+    xlim = _calc_lim_with_padding(feasible_x, padding_ratio, scale)
```

Comment:
Infeasible trials may be outside of feasible trials.

```suggestion
    xlim = _calc_lim_with_padding(feasible_x + infeasible_x, padding_ratio, scale)
```